backend/Tokens/managers.py

```diff
@@ -0,0 +1,39 @@
+from django.db import models
+from django.db import transaction
+from datetime import datetime
+
+class AuthenticatedGuestUserTokenManager(models.Manager):
+	def get_or_create_token(self, host_user, guest_user):
+		with transaction.atomic():
```

Comment:
Nice! In general, transaction scope should be as small as possible and only there when needed - aka there are at least two separate but co-dependent database / state changing calls to a separate system. As everything that can go wrong within a transaction scope will revert the changes, the bigger the transaction, more exceptions that will revert the changes. Not all exceptions / error cases necessarily mean that the db calls need to reverted to recover from the error. For the scalability I would:
1. Move this if else logic and transactional stuff to a manager.py and
2. Make sure the transaction scope is only wrapping the necessary parts - only inside the if statements that actually do 2 operations

Reply:
ok to the second point, but what do you mean with the first? This is a managers.py file.

Reply:
I removed the whole atomic transaction from the function as it didn't require it here.